api/TODO.adoc

```diff
@@ -0,0 +1,119 @@
+== Naming conventions
+
+Should we use plain names for classes/interfaces, using the package name to namespace them, or should we prefix them.
+
+`org.infinispan.api.sync.Cache`
+
+vs
+
+`org.infinispan.api.sync.SyncCache`
+
+== API dependencies
+
+The sync and async API must depend only on core JDK classes.
+The mutiny API has an optional dependency on mutiny.
+
+Should we depend on `javax.transaction` for the Transaction class or should we provide our own interface and a way to
```

Comment:
in an ideal world, I would say to remove `javax.transaction`. But I'm afraid that using conversion it would increase the code verbosity & complexity.

Reply:
We need to have a longer discussion about the JEE -> JakartaEE transition, but I should think this new API is an opportune time to make this transition.